node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -345,17 +355,30 @@ fn configure_database(
 }
 
 fn configure_cryptdes(
+    new_public_key: Option<bool>,
     persistent_config: &mut dyn PersistentConfiguration,
     db_password_opt: &Option<String>,
 ) -> CryptDEPair {
     let cryptde_pair = if let Some(db_password) = db_password_opt {
         let chain = Chain::from(persistent_config.chain_name().as_str());
-        let main_result = persistent_config.cryptde(db_password);
+        let main_result = match new_public_key {
+            None | Some(false) => persistent_config.cryptde(db_password),
+            Some(true) => {
+                let main_cryptde: Box<dyn CryptDE> = Box::new(CryptDEReal::new(chain));
+                persistent_config
+                    .set_cryptde(main_cryptde.as_ref(), db_password)
+                    .expect("Failed to set cryptde");
+                Ok(Some(main_cryptde))
+            }
+        };
         match main_result {
             Ok(Some(last_main_cryptde)) => {
                 CryptDEPair::new(last_main_cryptde, Box::new(CryptDEReal::new(chain)))
             }
             Ok(None) => {
+                if new_public_key == Some(false) {
+                    panic!("--new-public-key off: Cannot reestablish old public key: no old public key available");
+                }
```

Comment:
There was no `CryptDE` in the database. If the user is demanding we use the `CryptDE` from the database, we panic.